### ISSUE TITLE
vips from jpg to webp. no outputs. no errors.

### ISSUE THREAD

**Issue by kgdev** [jcupitt/libvips#676]
https://dzamqefpotdvf.cloudfront.net/p/images/072f52e4-e8d2-4ee0-b8db-4fde47e3ef05_07.jpg

Download the above image into local (issue1.jpg). And run

```
$ vips copy issue1.jpg issue1.webp
vips cache : jpegload filename="issue1.jpg" out=((VipsImage*) 0x7fd110803000) flags=((VipsForeignFlags) VIPS_FOREIGN_SEQUENTIAL) access=((VipsAccess) VIPS_ACCESS_SEQUENTIAL) -
vips cache : copy in=((VipsImage*) 0x7fd110803000) out=((VipsImage*) 0x7fd110803190) -
vips cache+: cast in=((VipsImage*) 0x7fd110803190) out=((VipsImage*) 0x7fd110803320) format=((VipsBandFormat) VIPS_FORMAT_UCHAR) -
vips cache+: linecache in=((VipsImage*) 0x7fd1108037d0) out=((VipsImage*) 0x7fd110803af0) tile-height=8 access=((VipsAccess) VIPS_ACCESS_SEQUENTIAL) -
vips cache+: sequential in=((VipsImage*) 0x7fd1108037d0) out=((VipsImage*) 0x7fd110803960) tile-height=8 -
```

No errors are shown. But the issue.webp is not generated.

**Comment by jcupitt** [comment]
Looks like it's a CMYK jpeg with an RGB ICC profile. The PNG writer reports an error:

```
$ vips copy url.jpeg x.png
(vips:13304): VIPS-WARNING **: profile 'icc': 'CMYK': invalid ICC profile color space
vips2png: unable to write "x.png"
```

But you're right, the webp one does not. I'll have a look.

**Comment by jcupitt** [comment]
Oh, the profile is correct, it's just not being applied. I'll fix it.

**Comment by kgdev** [comment]
Cool! Looking forward to the fix.

**Comment by jcupitt** [comment]
OK, should work now, I think. 

I think auto cmyk->rgb wasn't originally there since it would have ment we'd have to bundle an sRGB icc profile. But these days vips has it's own high-quality sRGB converter, so we can just use that. 

Anyway, a nice improvement, thanks for the suggestion!

**Comment by kgdev** [comment]
It seems it still doesn't work....

**Comment by kgdev** [comment]
It works great! We can close the issue now. Thank you very much!

**Comment by kgdev** [comment]
Can you release it?

**Comment by jcupitt** [comment]
I usually leave a few weeks between releases to avoid overloading downstream. Would it be OK to wait? There a few other fixes it would be nice to include in a bundle.

**Comment by kgdev** [comment]
Yes, it's ok to wait.

However, the color of the webp result is a title bit different than the original image. Could you please have a check?

**Comment by jcupitt** [comment]
I think the vips one is probably correct. What are you comparing it to?

I made some sample images:

http://www.rollthepotato.net/~john/ctest/

I made the images like this:

```
vips copy jpg-cmyk.jpg x.webp
vips copy jpg-cmyk.jpg x.png
vips icc_transform jpg-cmyk.jpg x.jpg ~/vips/share/nip2/data/sRGB.icm 
convert jpg-cmyk.jpg -intent Relative -profile ~/vips/share/nip2/data/sRGB.icm convert.png
```

Put the images into separate tabs and try flipping between them. `x.png` and `convert.png` look identical, the others seem to have slightly differing treatment of gamma, perhaps because of the compression, but colour looks the same.
